### PR TITLE
CI: Lint & Test

### PR DESCRIPTION
**Pull request by fvictorio**
Also: upgrade `solidity-comments-extractor`

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      - "*"
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 10
```
Comment (mattiaerre) [inline .github/workflows/CI.yml:18]: shall we use node 12 for this?

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          path: |
+            node_modules
+          key: ${{ runner.os }}-node-${{ matrix.node }}-${{ hashFiles('package-lock.json') }}
+      - name: Install
+        run: npm install
+      - name: lint
+        run: npm run lint
+
+  test_windows:
+    name: Test on Windows with Node 10
```
Comment (mattiaerre) [inline .github/workflows/CI.yml:32]: s/Node 10/Node 12

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        with:
+          node-version: 12
+      - uses: actions/checkout@v2
+      - name: Install
+        run: npm install
+      - name: Run tests
+        run: npm run test:all
+
+  test_macos:
+    name: Test on MacOS with Node 10
```
Comment (mattiaerre) [inline .github/workflows/CI.yml:45]: can we use Node 12 here too?

File: .github/workflows/CI.yml [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        run: npm install
+      - name: Run tests
+        run: npm run test:all
+
+  test_linux:
+    name: Test on Linux with Node ${{ matrix.node }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node: [10, 12, 14, 15]
```
Comment (mattiaerre) [inline .github/workflows/CI.yml:68]: I would just target 12, 14, and 15

**Review by mattiaerre — CHANGES_REQUESTED** [review]
this is great!

File: .github/workflows/CI.yml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 12
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        id: cache
+        with:
+          path: |
+            node_modules
+          key: ${{ runner.os }}-node-${{ matrix.node }}-${{ hashFiles('package-lock.json') }}
```
Comment (mattiaerre) [inline .github/workflows/CI.yml:25]: I do not see any `matrix.node` above

File: .github/workflows/CI.yml [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
```
Comment (mattiaerre) [inline .github/workflows/CI.yml:44]: I do not see any `matrix.node` above

File: .github/workflows/CI.yml [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+          node-version: ${{ matrix.node }}
+      - name: install vyper
+        run: docker pull ethereum/vyper:0.1.0b10
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        id: cache
+        with:
+          path: |
+            node_modules
+          key: ${{ runner.os }}-node-${{ matrix.node }}-${{ hashFiles('package-lock.json') }}
```
Comment (mattiaerre) [inline .github/workflows/CI.yml:68]: this `matrix.node` here makes sense to me

File: .github/workflows/CI.yml [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node: [12, 14, 15]
+    steps:
+      - uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node }}
+      - name: install vyper
+        run: docker pull ethereum/vyper:0.1.0b10
```
Comment (mattiaerre) [inline .github/workflows/CI.yml:61]: why do we need this docker image?

**Review by mattiaerre — COMMENTED** [review]
some more comments to you @fvictorio

**Review by mattiaerre — APPROVED** [review]
it looks like GH actions are running so I'm happy w/ this

File: .github/workflows/CI.yml (hunk 1 above)
Comment (fvictorio) [reply .github/workflows/CI.yml]: I copied this from the hardhat repo, we can't modify it. For reference:

- Node 10 will reach end of life in the coming weeks
- Node 11 and 13 are not maintained anymore
- Node 15 reaches end of life on June 1st

File: .github/workflows/CI.yml (hunk 8 above)
Comment (fvictorio) [reply .github/workflows/CI.yml]: Oops, error from the copy-paste.

**Comment by fvictorio** [comment]
Fixed the remaining points. @mattiaerre feel free to merge if you want!

**Comment by mattiaerre** [comment]
thanks @fvictorio I'll merge and publish a new version of the plugin right after

**Pull request merged**